pkgs/applications/editors/quartus-prime/default.nix

```diff
@@ -110,10 +138,40 @@ in buildFHSUserEnv {
     xorg.libXrender
   ];
 
-  extraInstallCommands = ''
+  passthru = {
+    inherit quartus;
+  };
+
+  extraInstallCommands = let
+    quartusExecutables = (map (c: "quartus/bin/quartus_${c}") [
+      "asm" "cdb" "cpf" "drc" "eda" "fit" "jbcc" "jli" "map" "pgm" "pow"
+      "sh" "si" "sim" "sta" "stp" "tan"
+    ]) ++ [ "quartus/bin/quartus" ];
+
+    qsysExecutables = (map (c: "quartus/sopc_builder/bin/qsys-${c}") [
+      "generate" "edit" "script"
+    ]);
+  in ''
     mkdir -p $out/share/applications
     cp ${desktopItem}/share/applications/* $out/share/applications
+
+    mkdir -p $out/quartus/bin
+    mkdir -p $out/quartus/sopc_builder/bin
+
+    WRAPPER=$out/bin/${name}
+    EXECUTABLES="${lib.concatStringsSep " " (quartusExecutables ++ qsysExecutables)}"
+
+    for executable in $EXECUTABLES; do
+        echo "#!${stdenv.shell}" >> $out/$executable
+        echo "$WRAPPER ${quartus}/$executable \$@" >> $out/$executable
+    done
+
+    cd $out
+    chmod +x $EXECUTABLES
+    ln --symbolic --relative --target-directory ./bin $EXECUTABLES
```

Comment:
I haven't tested this due to issues in adding the Quartus files I've downloaded to my store. I'm not sure I've managed to figure out what happens here eventually. Could you add some comments please?

Reply:
Sure. This just links all the executables into `$out/bin`, so if you do `nix run quartus-prime-lite` you get them added to your `$PATH`.

Reply:
I added a comment to this last bit, is that what was missing or does anything else need commenting as well?